File: backend/app/schemas/finance.py
from datetime import date
from decimal import Decimal
from typing import Optional

from pydantic import BaseModel


class CommissionResult(BaseModel):
    condominium_id: int
    condominium_name: str
    reference_month: str
    commission_type: str
    commission_value: Optional[Decimal]
    total_received: Decimal
    commission_due: Decimal


class OperatorPerformanceItem(BaseModel):
    condominium_id: int
    condominium_name: str
    total_billed: Decimal
    total_received: Decimal
    success_rate: Decimal
    commission_due: Decimal


class CommissionConfigUpdate(BaseModel):
    commission_type: str  # "fixed" | "percent" | "per_unit"
    commission_value: Optional[Decimal] = None


class CommissionRateCreate(BaseModel):
    product_id: int
    value_per_unit: Decimal
    valid_from: date


class CommissionRateOut(BaseModel):
    id: int
    condominium_id: int
    product_id: int
    product_name: Optional[str] = None
    value_per_unit: Decimal
    valid_from: date

    model_config = {"from_attributes": True}

File: frontend/src/stores/finance.ts
import { defineStore } from 'pinia'
import { ref } from 'vue'
import apiClient from '@/api/client'

export interface CommissionResult {
    condominium_id: number
    condominium_name: string
    reference_month: string
    commission_type: string
    commission_value: number | null
    total_received: number
    commission_due: number
}

export interface OperatorPerformanceItem {
    condominium_id: number
    condominium_name: string
    total_billed: number
    total_received: number
    success_rate: number
    commission_due: number
}

export interface CommissionRate {
    id: number
    condominium_id: number
    product_id: number
    product_name: string | null
    value_per_unit: number
    valid_from: string
}

export const useFinanceStore = defineStore('finance', () => {
    const commission = ref<CommissionResult | null>(null)
    const performance = ref<OperatorPerformanceItem[]>([])
    const commissionRates = ref<CommissionRate[]>([])
    const loading = ref(false)

    async function fetchCommission(condominiumId: number, month: string) {
        loading.value = true
        try {
            const { data } = await apiClient.get(`/finance/commission/${condominiumId}/${month}`)
            commission.value = data
        } finally {
            loading.value = false
        }
    }

    async function fetchPerformance(month: string) {
        loading.value = true
        try {
            const { data } = await apiClient.get(`/finance/operator-performance/${month}`)
            performance.value = data
        } finally {
            loading.value = false
        }
    }

    async function updateCommissionConfig(condominiumId: number, config: { commission_type: string; commission_value: number | null }) {
        const { data } = await apiClient.put(`/finance/condominiums/${condominiumId}/commission-config`, config)
        return data
    }

    async function fetchCommissionRates(condominiumId: number) {
        const { data } = await apiClient.get(`/finance/condominiums/${condominiumId}/commission-rates`)
        commissionRates.value = data
        return data
    }

    async function createCommissionRate(condominiumId: number, body: { product_id: number; value_per_unit: number; valid_from: string }) {
        const { data } = await apiClient.post(`/finance/condominiums/${condominiumId}/commission-rates`, body)
        commissionRates.value.unshift(data)
        return data
    }

    return {
        commission,
        performance,
        commissionRates,
        loading,
        fetchCommission,
        fetchPerformance,
        updateCommissionConfig,
        fetchCommissionRates,
        createCommissionRate,
    }
})
